### PR TITLE
feat(ai-label): add ts types

### LINKED ISSUE #17232: Add TypeScript types to AI Label (formerly Slug)

### PR DESCRIPTION
**Pull request by ariellalgilmore**
Closes #17232

Add typescript types to AI-label

#### Changelog

**Changed**

- Converted AILabel file from JS to TSX.
- updated align to copy `Popover` syntax with deprecated values

#### Testing / Reviewing

functionality stay the same
checks still passing
storybook looks good

### CODE REVIEW

**Comment by ariellalgilmore** [comment]
When checking the blame history in github you can still see the history even though it shows it as a new file

Screenshot 2024-08-28 at 8 21 25 AM

**Review by emyarod — APPROVED** [review]
looks good to me!

**Review by 2nikhiltom — APPROVED** [review]
LGTM 💯

**Review by annawen1 — APPROVED** [review]

File: packages/react/src/components/AILabel/index.tsx [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+    'top-right': 'top-end',
+    'bottom-left': 'bottom-start',
+    'bottom-right': 'bottom-end',
+    'left-bottom': 'left-end',
+    'left-top': 'left-start',
+    'right-bottom': 'right-end',
+    'right-top': 'right-start',
+  };
+  return mapping[deprecatedValue];
+};
```
Comment (tay1orjones) [inline packages/react/src/components/AILabel/index.tsx]: Not a blocker to merge at all, but at some point we should probably make this a reusable helper - we're going to be using this all over the codebase 😅

**Review by tay1orjones — APPROVED** [review]

File: packages/react/src/components/AILabel/index.tsx (hunk 1 above)
Comment (tay1orjones) [reply packages/react/src/components/AILabel/index.tsx]: I totally forgot we have one already: 

https://github.com/carbon-design-system/carbon/blob/afce28df13ae3475eb32c7b65e0599eb86775458/packages/react/src/tools/createPropAdapter.js#L44-L67

Comment (ariellalgilmore) [reply packages/react/src/components/AILabel/index.tsx]: ooh thanks! I updated AI label!

**Pull request merged**